stream/src/main/scala/org/apache/pekko/stream/stage/GraphStage.scala

```diff
@@ -1118,6 +1139,19 @@ abstract class GraphStageLogic private[stream] (val inCount: Int, val outCount:
     }
   }
 
+  private final class EmittingSpliterator[T](_out: Outlet[T], elems: Spliterator[T], _previous: OutHandler,
```

Comment:
* the advantage of a Spliterator over an iterator is that you can use trySplit() to partition the workload so you can get multiple threads to work on the data
* if you just use tryAdvance, you might as well be just using the standard iterator]
* so what do we gain here?

Reply:
you can avoid the `hasNext` and `next` call when you already have a `Spliterator`

Reply:
I'm not sure this will make a big diff to performance. I'll approve this but without jmh bechmarks, I am very sceptical about whether this extra code is worth the extra maintenance overhead.

Reply:
Thanks